public/app/features/explore/state/history.ts

```diff
@@ -79,20 +77,31 @@ export const addHistoryItem = (
   queries: DataQuery[]
 ): ThunkResult<void> => {
   return async (dispatch, getState) => {
-    const { richHistoryStorageFull, limitExceeded } = await addToRichHistory(
-      datasourceUid,
-      datasourceName,
-      queries,
-      false,
-      '',
-      !getState().explore.richHistoryStorageFull,
-      !getState().explore.richHistoryLimitExceededWarningShown
-    );
-    if (richHistoryStorageFull) {
-      dispatch(richHistoryStorageFullAction());
-    }
-    if (limitExceeded) {
-      dispatch(richHistoryLimitExceededAction());
+    // always add to local storage
+    getLocalRichHistoryStorage().addToRichHistory({
+      datasourceUid: datasourceUid,
+      datasourceName: datasourceName ?? '',
+      queries: queries,
+      starred: false,
+      comment: '',
+    });
+
+    if (config.queryHistoryEnabled) {
```

Comment:
Might be worth adding a comment why we do it only when `config.queryHistoryEnabled`.